pkgs/development/haskell-modules/make-package-set.nix

```diff
@@ -181,9 +181,37 @@ let
       };
     }) (self.callPackage src args);
 
+    # Creates a Haskell package from a source package by calling cabal2nix on the source.
+    callCabal2nixWithOptions' =
```

Comment:
I agree that it is nice to have this function.  I've wanted this functionality myself.

But would you be able to come up with a better name for `callCabal2nixWithOptions'`?  Maybe, `callCabal2nixConfigurable` (not that that is much better...)?